controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -52,14 +80,582 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephClient.", "CephClient", klog.KRef(r.cephClientRBDProvisioner.Namespace, r.cephClientRBDProvisioner.Name))
+		return err
+	}
+
+	return nil
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientRBDNode() error {
+
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "profile rbd",
+				"mgr": "allow rw",
+				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
+			},
+		},
+	}
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientRBDNode, func() error {
+		if err := r.own(r.cephClientRBDNode); err != nil {
+			return err
+		}
+
+		r.cephClientRBDNode.Spec = desired.Spec
+		return nil
+	})
+
+	cephClientRBDNodeDetails := &ocsv1alpha1.CephResourcesSpec{
+		Kind: r.cephClientRBDNode.Kind,
+		Name: r.cephClientRBDNode.Name,
+	}
+	if r.cephClientRBDNode.Status != nil {
+		cephClientRBDNodeDetails.Phase = string(r.cephClientRBDNode.Status.Phase)
+	}
+
+	r.updateCephResourceStatus(cephClientRBDNodeDetails)
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephClient.", "CephClient", klog.KRef(r.cephClientRBDNode.Namespace, r.cephClientRBDNode.Name))
+		return err
+	}
+
+	return nil
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientCephFSProvisioner() error {
+
+	// TODO: Update caps relavent to cephfs provisioner
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "allow r",
+				"mgr": "allow rw",
+				"osd": fmt.Sprintf("allow rw tag cephfs metadata=* path=%s", r.cephFilesystemSubVolumeGroup.Name),
+			},
+		},
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientCephFSProvisioner, func() error {
+		if err := r.own(r.cephClientCephFSProvisioner); err != nil {
+			return err
+		}
+
+		r.cephClientCephFSProvisioner.Spec = desired.Spec
+		return nil
+	})
+
+	cephClientCephFSProvisionerDetails := &ocsv1alpha1.CephResourcesSpec{
+		Kind: r.cephClientCephFSProvisioner.Kind,
+		Name: r.cephClientCephFSProvisioner.Name,
+	}
+	if r.cephClientCephFSProvisioner.Status != nil {
+		cephClientCephFSProvisionerDetails.Phase = string(r.cephClientCephFSProvisioner.Status.Phase)
+	}
+
+	r.updateCephResourceStatus(cephClientCephFSProvisionerDetails)
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephClient.", "CephClient", klog.KRef(r.cephClientCephFSProvisioner.Namespace, r.cephClientCephFSProvisioner.Name))
+		return err
+	}
+
+	return nil
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientCephFSNode() error {
+
+	// TODO: Update caps relavent to cephfs node
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "allow r",
+				"mgr": "allow rw",
+				"osd": fmt.Sprintf("allow rw tag cephfs *=* path=%s", r.cephFilesystemSubVolumeGroup.Name),
+				"mds": fmt.Sprintf("allow rw path=%s", r.cephFilesystemSubVolumeGroup.Name),
```

Comment:
Rook Uses 
```
"mon", "allow r",
"mgr", "allow rw",
"osd", "allow rw tag cephfs *=*",
"mds", "allow rw"
```
@leseb Can you check the cap changes done is correct for subvolume group integration

Reply:
@ajarr please confirm that the subvolume group integration (`path`) is correctly written. Thanks!